src/client/visualizations/base-visualization/base-visualization.tsx

```diff
@@ -0,0 +1,223 @@
+import * as React from 'react';
+import { $, ply, TimeRange, Expression, Dataset } from 'plywood';
+import {
+  MeasureModeNeeded,
+  Manifest,
+  Measure,
+  VisualizationProps,
+  DatasetLoad,
+  Splits,
+  DataSource,
+  Colors,
+  Resolve,
+  Essence
+} from '../../../common/models/index';
+
+import { SPLIT, SEGMENT, TIME_SEGMENT } from '../../config/constants';
+
+
+export interface BaseVisualizationState {
+  datasetLoad?: DatasetLoad;
+  dragStartTime?: Date;
+  dragTimeRange?: TimeRange;
+  roundDragTimeRange?: TimeRange;
+  dragOnMeasure?: Measure;
+  scrollLeft?: number;
+  scrollTop?: number;
+  hoverTimeRange?: TimeRange;
```

Comment:
Yeah, that's one thing I noticed. Do you think it makes sense to abstract this so both TimeSeries and, say, BarChart can use it ? Ideally, I'd like it to be whatever is hovered regardless of which vis is active, but I'm not sure this'll play well with TypeScript